src/__experimental__/components/date/date-picker.component.js

```diff
@@ -109,17 +109,25 @@ function getDisabledDays(minDate, maxDate) {
     return null;
   }
 
-  if (minDate) {
+  if (minDate && checkIsoFormatLength(minDate)) {
     days.push({ before: DateHelper.stringToDate(minDate) });
   }
 
-  if (maxDate) {
+  if (maxDate && checkIsoFormatLength(maxDate)) {
     days.push({ after: DateHelper.stringToDate(maxDate) });
   }
 
   return days;
 }
 
+function checkIsoFormatLength(date) {
+  if (date.length !== 10 || !DateHelper.isValidDate(date, { defaultValue: 'YYYY-MM-DD' })) {
+    return false;
+  }
+  const array = date.split('-');
+  return array.length === 3 && array[0].length === 4 && array[1].length === 2 && array[2].length;
```

Comment:
The existing code is fine, but for extra clarity, I think we could make this improvement:
```suggestion
  return array.length === 3 && array[0].length === 4 && array[1].length === 2 && array[2].length === 2;
```

Reply:
Good spot as always @kyanmichael, fixed in https://github.com/Sage/carbon/pull/2551/commits/bf2f8b513b1bbc673fd822391e033482513a09ed

Reply:
Thanks :white_check_mark: